text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,506 @@
+
+
+![Service Network](https://docs.aws.amazon.com/images/vpc-lattice/latest/ug/images/service-network.png)
+
+The construct `ServiceNetwork` provides for this. 
+
+
+The construct will implement `IServiceNetwork`.  
+
+
+```typescript
+/**
+ * Create a vpc lattice service network.
+ * Implemented by `ServiceNetwork`.
+ */
+e/**
+ * Create a vpc lattice service network.
+ * Implemented by `ServiceNetwork`.
+ */
+export interface IServiceNetwork extends core.IResource {
+
+  /**
+  * The Amazon Resource Name (ARN) of the service network.
+  */
+  readonly serviceNetworkArn: string;
+
+  /**
+   * The Id of the Service Network
+   */
+  readonly serviceNetworkId: string;
+  /**
+   * Grant Princopals access to the Service Network
+   */
+  grantAccessToServiceNetwork(principal: iam.IPrincipal[]): void;
+  /**
+   * Add Lattice Service Policy
+   */
+  addService(service: Service): void;
+  /**
+   * Associate a VPC with the Service Network
+   */
+  associateVPC(props: AssociateVPCProps): void;
+  /**
+   * Log To S3
+   */
+  logToS3(bucket: s3.Bucket | s3.IBucket ): void;
+  /**
+   * Send Events to Cloud Watch
+   */
+  sendToCloudWatch(log: logs.LogGroup | logs.ILogGroup ): void;
+  /**
+   * Stream to Kinesis
+   */
+  streamToKinesis(stream: kinesis.Stream | kinesis.IStream ): void;
+  /**
+   * Share the ServiceNetwork
+   */
+  share(props: ShareServiceNetworkProps): void;
+  /**
+   * Create and Add an auth policy to the Service Network
+   */
+  applyAuthPolicyToServiceNetwork(): void;
+}
+
+}
+``` 
+All of the class functions, apart from addAuthPolicy, are convience functions that mirror the properties. 
+
+`ServiceNetwork` will take `ServiceNetworkProps` as props
+```typescript
+/**
+ * The properties for the ServiceNetwork.
+ */
+export interface ServiceNetworkProps {
+
+  /** The name of the Service Network. If not provided Cloudformation will provide
+   * a name
+   * @default cloudformation generated name
+   */
+  readonly name?: string
+
+  /** The type of  authentication to use with the Service Network.
+   * @default 'AWS_IAM'
+   */
+  readonly authType?: AuthType | undefined;
+
+  /**
+   * S3 buckets for access logs
+   * @default no s3 logging
+   */
+
+  readonly s3LogDestination: s3.IBucket[] | undefined;
+  /**
+   * Cloudwatch Logs
+   * @default no logging to cloudwatch
+   */
+  readonly cloudwatchLogs?: logs.ILogGroup[] | undefined;
+
+  /**
+   * kinesis streams
+   * @default no streaming to Kinesis
+   */
+  readonly kinesisStreams?: kinesis.IStream[];
+
+  /**
+   * Lattice Services that are assocaited with this Service Network
+   * @default no services are associated with the service network
+   */
+  readonly services?: Service[] | undefined;
```

Comment:
Yes, and it should be an IService.

Services are attached to the Service network via AWS::VpcLattice::ServiceNetworkServiceAssociation. 

While a service network without any services or vpc associations is pretty useless, it is still valid. 

You may want to associate Services with the Service Network from other stacks, ( even in other accounts ).  So, I in this case, It should be optional, as it would be valid, to just build the service network in one stack, and then add vpc's and services later.